archive/compression/compression.go

```diff
@@ -139,3 +139,14 @@ func (compression *Compression) Extension() string {
 	}
 	return ""
 }
+
+// GetCompression returns the compression method of the the given reader, which is
+// returned by the method DecompressStream, or this method always return uncompressed.
+func GetCompression(r io.Reader) Compression {
+	if rw, ok := r.(*readCloserWrapper); ok {
```

Comment:
What if instead `readCloseWrapper` stored the `Compression` that was used? Otherwise this will have to be updated in tandem with `DecompressStream` (Assuming at some point `DecompressStream` supports more than gzip).